src/catalog/sequence_catalog.cpp

```diff
@@ -0,0 +1,300 @@
+ * @param   txn           current transaction
+ * @return  ResultType::SUCCESS if the sequence exists, ResultType::FAILURE
+ * otherwise.
+ * @exception throws SequenceException if the sequence already exists.
+ */
+bool SequenceCatalog::InsertSequence(oid_t database_oid,
+                                     std::string sequence_name,
+                                     int64_t seq_increment, int64_t seq_max,
+                                     int64_t seq_min, int64_t seq_start,
+                                     bool seq_cycle, type::AbstractPool *pool,
+                                     concurrency::TransactionContext *txn) {
+  LOG_DEBUG("Insert Sequence Database Oid: %u", database_oid);
+  LOG_DEBUG("Insert Sequence Sequence Name: %s", sequence_name.c_str());
+  if (GetSequence(database_oid, sequence_name, txn) != nullptr) {
+    throw SequenceException(
+        StringUtil::Format("Insert Sequence with Duplicate Sequence Name: %s",
+                           sequence_name.c_str()));
+  }
+
+  std::unique_ptr<storage::Tuple> tuple(
+      new storage::Tuple(catalog_table_->GetSchema(), true));
+
+  auto val0 = type::ValueFactory::GetIntegerValue(GetNextOid());
+  auto val1 = type::ValueFactory::GetIntegerValue(database_oid);
+  auto val2 = type::ValueFactory::GetVarcharValue(sequence_name);
+  auto val3 = type::ValueFactory::GetBigIntValue(seq_increment);
+  auto val4 = type::ValueFactory::GetBigIntValue(seq_max);
+  auto val5 = type::ValueFactory::GetBigIntValue(seq_min);
+  auto val6 = type::ValueFactory::GetBigIntValue(seq_start);
+  auto val7 = type::ValueFactory::GetBooleanValue(seq_cycle);
+  // When insert value, seqval = seq_start
+  auto val8 = type::ValueFactory::GetBigIntValue(seq_start);
+
+  tuple->SetValue(ColumnId::SEQUENCE_OID, val0, pool);
+  tuple->SetValue(ColumnId::DATABSE_OID, val1, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_NAME, val2, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_INC, val3, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_MAX, val4, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_MIN, val5, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_START, val6, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_CYCLE, val7, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_VALUE, val8, pool);
+
+  // Insert the tuple
+  return InsertTuple(std::move(tuple), txn);
+}
+
+/* @brief   Delete the sequence by name.
+ * @param   database_oid  the databse_oid associated with the sequence
+ * @param   sequence_name the name of the sequence
+ * @param   txn           current transaction
+ * @return  ResultType::SUCCESS if the sequence exists, ResultType::FAILURE
+ * otherwise.
+ */
+ResultType SequenceCatalog::DropSequence(const std::string &database_name,
+                                         const std::string &sequence_name,
+                                         concurrency::TransactionContext *txn) {
+  if (txn == nullptr) {
+    LOG_TRACE("Do not have transaction to drop sequence: %s",
```

Comment:
Logging level of LOG_TRACE may be too low here. May consider switching to a higher level, such as LOG_INFO.